Public/Src/Cache/ContentStore/Library/Stores/ContentHashAddressableSnapshot.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using BuildXL.Cache.ContentStore.Hashing;
+
+namespace BuildXL.Cache.ContentStore.Stores
+{
+    /// <summary>
+    /// This class is specifically for usage within <see cref="ContentHashAddressableSnapshot{T}"/>. It is a reference type on purpose, in order to avoid issues
+    /// with the maximum object size limit.
+    /// </summary>
+    /// <typeparam name="T">Type tagged with a hash</typeparam>
+    public class PayloadFromDisk<T> : IComparable<PayloadFromDisk<T>>

```

Comment:
Is it possible to make internal? (I know our code base is bad about this and tends to just make everything public, but I would like to change if possible.)

Reply:
Making this internal implies that the Reconstruct interface would also become internal, and that will cascade to MemoryContentDirectory, which I will probably keep on cascading as it is public.

I can definitely do this if you want, but I think it will be a big enough amount of changes that it'd be better to do in a different PR, so as not to side-track this one.

Reply:
Understood. No need to change, just wanted to make sure we aren't doing so unnecessarily.

---
In reply to: [285277556](https://github.com/microsoft/BuildXL/pull/329#discussion_r285277556) [](ancestors = 285277556)